src/Database/Kioku/Internal/TrieIndex.hs

```diff
@@ -387,3 +434,19 @@ breakCommonPrefix b1 b2 =
      , BS.drop prefixLength b1
      , BS.drop prefixLength b2
      )
+
+breakCommonPrefixMultiKey :: MultiKey
+                          -> BS.ByteString
+                          -> (BS.ByteString, MultiKey, BS.ByteString)
+breakCommonPrefixMultiKey (MultiKey prefix kids) arc =
+  case breakCommonPrefix prefix arc of
+    (common, "", "") -> (common, MultiKey "" kids, "")
+    (common, remPrefix, "") -> (common, MultiKey remPrefix kids, "")
```

Comment:
It looks like we could actually drop the first branch of this `case` because the second one is just a more general version of it.